src/components/tx/SignOrExecuteForm/hooks.ts

```diff
@@ -138,3 +140,29 @@ export const useIsExecutionLoop = (): boolean => {
   const { safeAddress } = useSafeInfo()
   return wallet ? sameString(wallet.address, safeAddress) : false
 }
+
+export const useRecommendedNonce = (
+  safeTx?: SafeTransaction,
+  initialNonce?: number,
+):
+  | {
+      nonce: number
+      safeTxGas: number
+    }
+  | undefined => {
+  const { safe } = useSafeInfo()
+
+  const [data] = useAsync(() => {
+    if (!safeTx) return
+    return getRecommendedTxParams(safeTx)
+  }, [safeTx, safe?.txQueuedTag])
+
+  return useMemo(() => {
+    return data
+      ? {
+          nonce: initialNonce !== undefined && initialNonce !== -1 ? initialNonce : data.nonce,
```

Comment:
Nit: For readability I would let `useRecommendedNonce` always return the recommended one and react to an already initially set nonce in a separate step, which overrides the txs nonce if no `initialNonce` was set.

Reply:
I initially had it like this, but this introduced the same not-so-nice condition in SignOrExecute...